dsl/src/main/scala/org/http4s/dsl/Path.scala

```diff
@@ -31,11 +32,8 @@ object Path {
       Path("/" + str)
     else {
       val slash = str.lastIndexOf('/')
-      val prefix = Path(str.substring(0, slash))
-      if (slash == str.length - 1)
-        prefix
-      else
-        prefix / str.substring(slash + 1)
+      val prefix = Path(UrlCodingUtils.urlDecode(str.substring(0, slash)))
```

Comment:
You're right, there is a bug there. I'll add a test and fix it. Thanks for
the review!